internal/backends/winit/renderer/femtovg/glcontext.rs

```diff
@@ -49,144 +51,108 @@ impl OpenGLContext {
-                        )?
-                    };
-
-                    let attrs = SurfaceAttributesBuilder::<WindowSurface>::new().build(
-                        _window.raw_window_handle(),
-                        width,
-                        height,
-                    );
-
-                    let surface =
-                        unsafe { config.display().create_window_surface(&config, &attrs)? };
-
-                    Ok((surface, not_current_gl_context))
-                };
-
-            let num_prefs = prefs.len();
-            let (surface, not_current_gl_context) = prefs
-                .into_iter()
-                .enumerate()
-                .find_map(|(i, pref)| {
-                    let is_last = i == num_prefs - 1;
-
-                    match try_create_surface(pref) {
-                        Ok(result) => Some(result),
-                        Err(glutin_error) => {
-                            if is_last {
-                                panic!("Glutin error creating GL surface: {}", glutin_error);
-                            }
-                            None
-                        }
+    #[cfg(not(target_arch = "wasm32"))]
+    pub fn new_context<T>(
+        window_builder: winit::window::WindowBuilder,
+        window_target: &winit::event_loop::EventLoopWindowTarget<T>,
+    ) -> Result<(winit::window::Window, Self), Box<dyn std::error::Error>> {
+        let config_template_builder = glutin::config::ConfigTemplateBuilder::new();
+
+        let (window, gl_config) = glutin_winit::DisplayBuilder::new()
+            .with_preference(glutin_winit::ApiPrefence::FallbackEgl)
+            .with_window_builder(Some(window_builder.clone()))
+            .build(window_target, config_template_builder, |it| {
+                it.reduce(|accum, config| {
+                    let transparency_check = config.supports_transparency().unwrap_or(false)
+                        & !accum.supports_transparency().unwrap_or(false);
+
+                    if transparency_check || config.num_samples() < accum.num_samples() {
+                        config
+                    } else {
+                        accum
                     }
                 })
-                .unwrap();
-
-            #[cfg(target_os = "macos")]
-            if let raw_window_handle::RawWindowHandle::AppKit(
-                raw_window_handle::AppKitWindowHandle { ns_view, .. },
-            ) = _window.raw_window_handle()
-            {
-                use cocoa::appkit::NSView;
-                let view_id: cocoa::base::id = ns_view as *const _ as *mut _;
-                unsafe {
-                    NSView::setLayerContentsPlacement(view_id, cocoa::appkit::NSViewLayerContentsPlacement::NSViewLayerContentsPlacementTopLeft)
-                }
-            }
+                .expect("internal error: Could not find any matching GL configuration")
```

Comment:
This only panics if the iterator was empty.  I guess this cannot really happen?

Reply:
I think theoretically this could happen, but the glutin-winit API has no way of handling this at the moment. All users of this `build()` function call unwrap (even the glutin example).

In practice I think that if the egl/glx/etc. display initialisation succeeds, there must be _some_ available configurations. Otherwise it's pretty useless of the driver to offer that. So I think the case we should handle is to make sure that the display initialisation succeeds, that's earlier up.